Update WildFly Common to 2.0.1 and adjust substitutions

gsmet opened the pull request:
@dmlloyd adjusted WildFly Common to rely on SmallRye Common in 2.0.x but we never updated to this version which was a mistake.

Also adjusted the substitutions.

gsmet commented:
I tested a few modules that might rely on WildFly Common and things were good, we will know more once CI has passed.

dmlloyd commented on .github/dependabot.yml:
```diff
@@ -40,6 +40,7 @@ updates:
       - dependency-name: io.quarkus.*:*
       - dependency-name: io.quarkus:*
       # Elytron
+      - dependency-name: org.wildfly.common:*
```
This should be before the `# Elytron` comment, which is referring to `org.wildfly.security`.

gsmet replied:
Yeah, I thought about changing it to WildFly and then forgot. I moved it in the misc section at the top.

dmlloyd reviewed (approved):
LGTM (other than one tiny detail). Good catch, sorry I missed it earlier.

gsmet commented:
Hmmm, actually, I will push an update soon as we still need the substitutions to be perfectly safe.

gsmet committed:
Update WildFly Common to 2.0.1 and adjust substitutions

@dmlloyd adjusted WildFly Common to rely on SmallRye Common in 2.0.x but
we never updated to this version which was a mistake.

Also adjust the substitutions. We keep them around until the deprecated
static elements are dropped.

gsmet commented:
@dmlloyd in the end, I restored (and adjusted the subtitutions) instead as I think they are still useful in the end.

gsmet commented:
@ozangunalp see the flaky tests above.

ozangunalp commented:
@gsmet yes I am looking at them since yesterday, I couldn't reproduce it yet. Will disable or remove them if I can't find why they are failing.

gsmet commented:
@ozangunalp wouldn't it be a sign of a race condition? Which could be problematic in the case of propagating the context.
